persistence/sql/identity/persister_identity.go

```diff
@@ -824,14 +824,8 @@ func (p *IdentityPersister) ListIdentities(ctx context.Context, params identity.
 		identifier := params.CredentialsIdentifier
 		identifierOperator := "="
 		if identifier == "" && params.CredentialsIdentifierSimilar != "" {
-			identifier = params.CredentialsIdentifierSimilar
-			identifierOperator = "%"
-			switch con.Dialect.Name() {
-			case "postgres", "cockroach":
-			default:
-				identifier = "%" + identifier + "%"
-				identifierOperator = "LIKE"
-			}
+			identifier = strings.ReplaceAll(strings.ReplaceAll(strings.ReplaceAll(params.CredentialsIdentifierSimilar, "\\", "\\\\"), "%", "\\%"), "_", "\\_") + "%"
```

Comment:
What is this for? Wouldn't this be enough?

```
identifier = identifier + "%"
```

Reply:
Well, if you put `%foo` as the parameter, you will be able to search for `%foo%`, which is not covered by an index. So we have to escape all pattern characters, and the escape character.

Reply:
Can you please add one test to ensure this escaping works correctly?